hack/setup_env.sh

```diff
@@ -4,45 +4,62 @@ set -o nounset
 set -o pipefail
 set -o errexit
 set -o xtrace
+set -o verbose
+
+if [[ $(id -u) -eq 0 ]]; then
+        BINDIR="/usr/local/bin"
+elif command -v systemd-detect-virt; then
+    if systemd-detect-virt -cq; then
+        if [ -n "${TOOLBOX_PATH+x}" ]; then
+            BINDIR="$(systemd-path user-binaries)"
+        fi
+    else
+        BINDIR="$(systemd-path user-binaries)"
+    fi
+else
+    BINDIR="/usr/local/bin"
+fi
 
 function print_help() {
   ALL_FUNCS="kustomize|golang|assisted_service|hive_from_upstream|print_help"
   echo "Usage: bash ${0} (${ALL_FUNCS})"
 }
 
 function kustomize() {
-  if which kustomize; then
-    return
+  if ! type -f kustomize; then
+    # We tried using "official" install_kustomize.sh script, but it used too much rate-limited APIs of GitHub
+    curl -L --retry 5 "https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.3.0/kustomize_v4.3.0_linux_amd64.tar.gz" | \
+    tar -zx -C "$BINDIR"
   fi
-
-  # We tried using "official" install_kustomize.sh script, but it used too much rate-limited APIs of GitHub
-  curl -L --retry 5 "https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.3.0/kustomize_v4.3.0_linux_amd64.tar.gz" | \
-    tar -zx -C /usr/bin/
 }
 
 function golang() {
   echo "Installing golang..."
-  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o /tmp/golang_installer
-  chmod u+x /tmp/golang_installer
-  /tmp/golang_installer
+  local tmpdir="$(mktemp -d)"
+  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o "${tmpdir}/golang_installer"
+  chmod u+x ${tmpdir}/golang_installer
+  "${tmpdir}/golang_installer"
   rm /tmp/golang_installer
+  rmdir "$tmpdir"
 
   echo "Activating go command on current shell..."
   set +u
-  source /root/.bash_profile
+  source "${HOME}/.bash_profile"
```

Comment:
This reminds me that I have a pending PR for the `deploy/operator` scripts to do the same.

Reply:
Oops! I would have sworn I did this part. Or maybe it was the one that made me decide to finish it in the morning